tests/end_to_end/workflow/internal_loop.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2020-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from openfl.experimental.workflow.interface.fl_spec import FLSpec
+from openfl.experimental.workflow.interface.participants import Aggregator, Collaborator
+from openfl.experimental.workflow.runtime import LocalRuntime
+from openfl.experimental.workflow.placement.placement import aggregator, collaborator
+import numpy as np
+pass
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger(__name__)
+
+class TestFlowInternalLoop(FLSpec):
```

Comment:
We can add the purpose of the Flow (missing in original test cases): Testflow to validate the Internal Loop functionality